src/org/zaproxy/zap/extension/formhandler/ExtensionFormHandler.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.formhandler;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import java.util.List;
+
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.extension.ExtensionHook;
+
+import org.zaproxy.zap.extension.spider.ExtensionSpider;
+import org.zaproxy.zap.model.DefaultValueGenerator;
+
+public class ExtensionFormHandler extends ExtensionAdaptor {
+
+    public static final String NAME = "ExtensionFormHandler";
+
+    protected static final String PREFIX = "formhandler";
+
+    private FormHandlerParam param;
+
+    private OptionsFormHandlerPanel optionsFormHandlerPanel;
+
+    public ExtensionFormHandler() {
+        super(NAME);
+    }
+
+    @Override
+    public void hook(ExtensionHook extensionHook) {
+        super.hook(extensionHook);
+
+        extensionHook.addOptionsParamSet(getParam());
+        ExtensionSpider extension = (ExtensionSpider) Control.getSingleton().getExtensionLoader()
+                .getExtension(ExtensionSpider.NAME);
+        if (extension != null){
+            extension.setValueGenerator(new FormHandlerValueGenerator(getParam()));
+        }
+
+        if (getView() != null) {
+            extensionHook.getHookView().addOptionPanel(getOptionsFormHandlerPanel());
+        }
+    }
+
+    @Override
+    public boolean canUnload() {
+        // The extension can be dynamically unloaded, all resources used/added can be freed/removed from core.
+        return true;
+    }
+
+    @Override
+    public void unload() {
+        super.unload();
+        ExtensionSpider extension = (ExtensionSpider) Control.getSingleton().getExtensionLoader()
```

Comment:
We also need to check that's not `null` before using it here, like done in the `hook` method. BTW, it could use `getExtension(ExtensionSpider.class);` instead of `getExtension(ExtensionSpider.NAME);` which does not require the cast (same change could be done in `hook`).